ci_tools/check_slots.py

```diff
@@ -26,82 +28,132 @@
+                    first_iteration = False
+                output_dict['summary'] += f"\t-{value['message']}\n\n"
+                output_dict['annotations'].append(value['annotations'])
+    return output_dict
+
+def fill_dictionary(title, message, file, start_line, end_line, annotation_level, annotation_msg):
+    """
+    Fills a dictionary with the provided data.
+
+    Args:
+        data (dict): The data to populate the dictionary. It should be a dictionary
+                     where the keys represent the dictionary keys, and the values
+                     represent the corresponding values.
+
+    Returns:
+        dict: A dictionary filled with the provided data.
+    """
+    filled_dict = {
+        'title':title,
+        'message':message,
+        'annotations':{
+            'path':file,
+            'start_line':start_line,
+            'end_line':end_line,
+            'annotation_level':annotation_level,
+            'message':annotation_msg
+        }
+    }
+    return filled_dict
+
 for mod_name in ast_modules:
     mod = importlib.import_module('pyccel.ast.'+mod_name)
     all_attr = getattr(mod, '__all__', None)
     if all_attr:
         sorted_all = list(all_attr)
         sorted_all.sort()
         if sorted_all != list(all_attr):
-            non_alphabetical_all.append(mod_name)
+            lines = inspect.getsource(mod).splitlines()
+            start_line = -1
+            end_line = -1
+            for line_num, line in enumerate(lines):
+                if '__all__' in line:
+                    start_line = line_num + 1
+                    while ')' not in line:
+                        line_num += 1
+                        line = lines[line_num]
+                        end_line = line_num + 1
+                    ErrorCollection['non_alphabetical_all'].append(fill_dictionary("Non-alphabetical `__all__`", f"`pyccel.ast.{mod_name}`",
+                        inspect.getfile(mod), start_line, end_line, "warning", f"Sort the __all__ attribute of `{mod_name}`"))
+                    break
     else:
-        missing_all.append(mod_name)
+        ErrorCollection['missing_all'].append(fill_dictionary("Missing `__all__`", f"`pyccel.ast.{mod_name}`",
+            inspect.getfile(mod), 1, 1, "failure", f"Missing __all__ attribute in: `{mod_name}`"))
 
     classes = inspect.getmembers(mod, inspect.isclass)
+    first_iteration = True
     for cls_name, cls_obj in classes:
         if inspect.getmodule(cls_obj) is not mod:
             continue
         super_classes = cls_obj.mro()[1:]
         if '__slots__' not in cls_obj.__dict__:
-            missing_slots.append(f"{mod_name}.{cls_name}")
+            ErrorCollection['missing_slots'].append(fill_dictionary("Classes with no `__slots__`", f"`pyccel.ast.{mod_name}.{cls_name}`",
+                inspect.getfile(mod), inspect.getsourcelines(cls_obj)[1], inspect.getsourcelines(cls_obj)[1], "failure", f"`{mod_name}.{cls_name}` classe with no `__slots__`"))
```

Comment:
`inspect.getsourcelines(cls_obj)[1]` seems to be duplicated. Is this necessary? If so can we save the value in a variable instead of recalculating it